File: src/auth/browser.rs

pub fn try_open(url: &str) -> bool {
    #[cfg(target_os = "macos")]
    {
        return std::process::Command::new("open")
            .arg(url)
            .spawn()
            .is_ok();
    }

    #[cfg(target_os = "linux")]
    {
        return std::process::Command::new("xdg-open")
            .arg(url)
            .spawn()
            .is_ok();
    }

    #[cfg(target_os = "windows")]
    {
        return std::process::Command::new("cmd")
            .args(["/C", "start", "", url])
            .spawn()
            .is_ok();
    }

    #[cfg(not(any(target_os = "macos", target_os = "linux", target_os = "windows")))]
    {
        let _ = url;
        false
    }
}

File: src/kernel/std/browser.ts
import { getPlatform } from "@/kernel/std/proc/platform.ts";

export async function openBrowser(url: string): Promise<void> {
  const platform = getPlatform();
  try {
    if (platform === "macos") {
      // `-u` treats the arg as a URL (avoids path/option misparse on long OAuth URLs).
      Bun.spawn(["open", "-u", url], { stdout: "ignore", stderr: "ignore" });
      return;
    }
    if (platform === "windows") {
      Bun.spawn(["rundll32.exe", "url.dll,FileProtocolHandler", url], {
        stdout: "ignore",
        stderr: "ignore",
      });
      return;
    }
    Bun.spawn(["xdg-open", url], { stdout: "ignore", stderr: "ignore" });
  } catch {}
}
